File: frontend/src/features/coordinacion/types/coloquios.ts
export interface Convocatoria {
  id: string
  materia_id: string
  materia_nombre: string
  cohorte_id: string
  cohorte_nombre: string
  tipo: string
  instancia: string
  dias_disponibles: number
  total_convocados: number
  reservas_activas: number
  total_reservas: number
  total_resultados: number
  cupos_libres: number
  creada_at: string
}

export interface ConvocatoriaDetalle {
  id: string
  materia_id: string
  materia_nombre: string
  cohorte_id: string
  cohorte_nombre: string
  tipo: string
  instancia: string
  dias_disponibles: number
  alumnos: ConvocadoAlumno[]
  reservas: Reserva[]
  resultados: Resultado[]
}

export interface ConvocadoAlumno {
  id: string
  nombre: string
  apellidos: string
  email: string
}

export interface Reserva {
  id: string
  alumno_id: string
  alumno_nombre: string
  fecha_hora: string
  estado: string
}

export interface Resultado {
  id: string
  alumno_id: string
  alumno_nombre: string
  nota_final: string
}

export interface ConvocatoriaPayload {
  materia_id: string
  cohorte_id: string
  tipo: string
  instancia: string
  dias_disponibles: number
}

export interface ResultadoPayload {
  alumno_id: string
  nota_final: string
}

File: backend/app/schemas/coloquios.py
"""Pydantic schemas for Coloquios (Evaluacion, ReservaEvaluacion, ResultadoEvaluacion)."""

from datetime import date

from pydantic import BaseModel, ConfigDict


class EvaluacionCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    materia_id: str
    cohorte_id: str
    instancia: str
    tipo: str = "Coloquio"
    dias_disponibles: int = 5
    cupo_por_dia: int = 10


class EvaluacionRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    materia_id: str
    cohorte_id: str
    tipo: str
    instancia: str
    dias_disponibles: int
    cupo_por_dia: int
    activa: bool
    convocados: list
    materia_nombre: str | None = None
    cohorte_nombre: str | None = None


class EvaluacionDetalle(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    materia_id: str
    cohorte_id: str
    tipo: str
    instancia: str
    dias_disponibles: int
    cupo_por_dia: int
    activa: bool
    convocados: list
    materia_nombre: str | None = None
    cohorte_nombre: str | None = None
    total_convocados: int = 0
    reservas_activas: int = 0
    cupos_libres: int = 0


class ReservaCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    fecha: date
    hora: str


class ReservaRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    evaluacion_id: str
    alumno_id: str
    fecha: date
    hora: str
    estado: str


class ResultadoCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    alumno_id: str
    nota_final: str


class ResultadoRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    evaluacion_id: str
    alumno_id: str
    nota_final: str


class AlumnosImport(BaseModel):
    model_config = ConfigDict(extra="forbid")

    alumno_ids: list[str]


class AdminGlobalItem(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    materia_id: str
    cohorte_id: str
    instancia: str
    tipo: str
    activa: bool
    total_convocados: int = 0
    reservas_activas: int = 0
    total_reservas: int = 0
    total_resultados: int = 0
